src/js/socketRepl.js

```diff
@@ -7,29 +7,48 @@ import { prompt, processLine } from './repl';
 
 
 let socketServer: ?net$Server = null;
+export const sockets: { [id: string]: net$Socket } = {};
 
 export function getSocketServer(): ?net$Server {
   return socketServer;
 }
 
 export function handleConnection(socket: net$Socket): readline$Interface {
-  const rl = readline.createInterface({ input: socket, output: socket, terminal: true });
-  rl.write(createBanner());
-  prompt(rl, false, 'cljs.user');
+  const socketId = `${socket.remoteAddress || ''}:${socket.remotePort}`;
+  socket.on('close', () => delete sockets[socketId]);
+  sockets[socketId] = socket;
+
+  const rl = readline.createInterface({ input: socket, output: socket });
   rl.on('line', (line: string) => processLine(rl, line));
-  rl.on('SIGINT', () => socket.end('Goodbye!'));
-  return rl;
-}
+  rl.on('SIGINT', () => socket.destroy());
 
-export function open(port: number, host?: string): void {
-  socketServer = net.createServer((socket: net$Socket) => handleConnection(socket));
-  socketServer.listen(port, host);
+  // $FlowIssue - output missing from readline$Interface
+  rl.output.write(createBanner());
+  prompt(rl, false, 'cljs.user');
+  return rl;
 }
 
 export function close(): void {
   if (!socketServer) {
     return;
   }
 
-  socketServer.close();
+  Object.keys(sockets)
+    .forEach((socketId: string) => {
+      try {
+        sockets[socketId].destroy();
+      } catch (e) {} // eslint-disable-line no-empty
+    });
+
+  socketServer.close(() => process.exit());
+}
+
+export function open(port: number, host?: string): void {
+  socketServer = net.createServer((socket: net$Socket) => handleConnection(socket));
+  socketServer.listen(port, host);
+
+  process.on('SIGTERM', close);
+  process.on('SIGINT', close);
```

Comment:
Not sure about this one. does this mean that if you hit `ctrl+C` the socket REPL is closed? That might not be in line with the TTY REPL, where we allow users to press `ctrl+C` (see  [here](https://github.com/anmonteiro/lumo/blob/8ab787834d0d5abbc2ffae229dac5ef54591fb82/src/js/repl.js#L86))